purchases/src/androidTestIntegrationTest/kotlin/com/revenuecat/purchases/PurchasesIntegrationTest.kt

```diff
@@ -0,0 +1,197 @@
+    }
+
+    @Test
+    fun customerInfoCanBeFetchedFromBackendAndThenGottenFromCache() {
+        val lock = CountDownLatch(1)
+
+        activityScenarioRule.scenario.onActivity {
+            Purchases.sharedInstance.getCustomerInfoWith(
+                CacheFetchPolicy.FETCH_CURRENT,
+                onError = {
+                    fail("fetching from backend should be success. Error: ${it.message}")
+                },
+                onSuccess = { fetchedCustomerInfo ->
+                    Purchases.sharedInstance.getCustomerInfoWith(
+                        CacheFetchPolicy.CACHE_ONLY,
+                        onError = {
+                            fail("fetching from cache should be success. Error: ${it.message}")
+                        },
+                        onSuccess = { cachedCustomerInfo ->
+                            assertThat(cachedCustomerInfo).isEqualTo(fetchedCustomerInfo)
+                            lock.countDown()
+                        }
+                    )
+                }
+            )
+        }
+        lock.await(testTimeout.inWholeSeconds, TimeUnit.SECONDS)
+        assertThat(lock.count).isZero
+    }
+
+    @Test
+    fun canFetchOfferings() {
+        val lock = CountDownLatch(1)
+
+        val storeProduct = StoreProductFactory.createStoreProduct()
+        mockBillingAbstract.mockQuerySkuDetails(querySkuDetailsSubsReturn = listOf(storeProduct))
+
+        onActivityReady {
+            Purchases.sharedInstance.getOfferingsWith(
+                onError = { error -> fail("Get offerings should be successful. Error: ${error.message}") },
+                onSuccess = { offerings ->
+                    assertThat(offerings.current).isNotNull
+                    assertThat(offerings.current?.availablePackages?.size).isEqualTo(1)
+                    assertThat(offerings.current?.monthly?.product?.sku).isEqualTo("monthly_intro_pricing_one_week")
+
+                    lock.countDown()
+                }
+            )
+        }
+        lock.await(testTimeout.inWholeSeconds, TimeUnit.SECONDS)
+        assertThat(lock.count).isZero
+    }
+
+    @Test
+    fun canPurchaseSubsProduct() {
+        val lock = CountDownLatch(1)
+
+        val storeProduct = StoreProductFactory.createStoreProduct()
+        val storeTransaction = StoreTransactionFactory.createStoreTransaction()
+        mockBillingAbstract.mockQuerySkuDetails(querySkuDetailsSubsReturn = listOf(storeProduct))
+
+        onActivityReady { activity ->
+            Purchases.sharedInstance.purchaseProductWith(
+                activity = activity,
+                storeProduct = storeProduct,
+                onError = { error, _ -> fail("Purchase should be successful. Error: ${error.message}") },
+                onSuccess = { transaction, customerInfo ->
+                    assertThat(transaction).isEqualTo(storeTransaction)
+                    assertThat(customerInfo.allPurchaseDatesByProduct.containsKey(storeProduct.sku)).isTrue
```

Comment:
Since the entitlement is not active after a while (being a test purchase), I can't test the fields related to active entitlements, though it would be nice...